homeassistant/config_entries.py

```diff
@@ -2814,6 +2820,36 @@ def async_update_reload_and_abort(
                 reason = "reconfigure_successful"
         return self.async_abort(reason=reason)
 
+    @callback
+    def async_show_form(
+        self,
+        *,
+        step_id: str | None = None,
+        data_schema: vol.Schema | None = None,
+        errors: dict[str, str] | None = None,
+        description_placeholders: Mapping[str, str | None] | None = None,
+        last_step: bool | None = None,
+        preview: str | None = None,
+    ) -> ConfigFlowResult:
+        """Return the definition of a form to gather user input.
+
+        The step_id parameter is deprecated and will be removed in a future release.
+        """
+        if self.source == SOURCE_REAUTH:
+            # If the integration does not provide a name for the reauth title,
+            # we append it to the description placeholders.
```

Comment:
As we have seen in weird implementations, I think this might be a breaking change for reauth flows triggered without a config entry.

We may need to ensure that `entry_id` is available.
```suggestion
        if self.source == SOURCE_REAUTH and "entry_id" in self.context:
            # If the integration does not provide a name for the reauth title,
            # we append it to the description placeholders.
            # We also need to check entry_id as some integrations bypass the 
            # reauth helpers and create a flow without it
```

Reply:
Fair point even though I think it should be mandatory to provide an `entry_id` for both `reauth` and `reconfigure` flows.
Anyhow maybe that check should go elsewhere and do some logging (deprecation) first.

Reply:
I agree - a deprecation is needed in the "init".

But I think the check is still needed here - at least until the deprecation ends.

Reply:
Unfortunately yes, anyway will add it once I rebase over https://github.com/home-assistant/core/pull/129313 that's pending